### PR TITLE
Detect missing SerializedName annotation in Retrofit request body

### PR DESCRIPTION
**Pull request by hbmartin**
- Applies same missing annotation field logic as return type for an `@Body` parameter in a Retrofit request
- Adds and configures Detekt for static analysis (with baseline for current warning)
- Minor version upgrades for Kotlin, Gradle, and Lint
- Some minor formatting changes for Detekt

### CODE REVIEW

File: checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassSerializedNameDetector.kt [hunk 1]
```diff
@@ -22,40 +22,58 @@ internal class NetworkLayerClassSerializedNameDetector : RetrofitReturnTypeDetec
...
-                        location = context.getNameLocation(node),
-                        message = "Return type doesn't have @SerializedName annotation for $nonFinalFields fields."
+                    issue = ISSUE_NETWORK_LAYER_CLASS_SERIALIZED_NAME_RULE,
+                    scopeClass = node,
+                    location = context.getNameLocation(node),
+                    message = "Return type doesn't have @SerializedName annotation for $nonFinalFields fields."
                 )
             }
+
+            findAllBodyParameterTypeFieldsOf(node)?.let { (classType, fieldsList) ->
```
Comment (hbmartin) [inline checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassSerializedNameDetector.kt:39]: New functionality in this class, the rest is just formatting

File: build.gradle [hunk 2]
```diff
@@ -20,6 +21,9 @@ buildscript {
 
 plugins {
     id("com.gradleup.auto.manifest") version "1.0.4"
+    id("io.gitlab.arturbosch.detekt") version "1.20.0"
+    id("org.jlleitschuh.gradle.ktlint") version "10.2.1"
+    id("com.github.ben-manes.versions") version "0.42.0"
```
Comment (kozaxinan) [inline build.gradle:26]: thank you for this integrations 💯

**Review by kozaxinan — APPROVED** [review]
If you can add a couple of test cases, that would be great. 
Thank you for contribution.

File: checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassSerializedNameDetector.kt (hunk 1 above)
Comment (kozaxinan) [reply checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassSerializedNameDetector.kt:39]: This is useful, can you also add a couple of test cases for this new feature?

File: build.gradle (hunk 2 above)
Comment (kozaxinan) [reply build.gradle:26]: it would be nice to run detect in github action as well.

**Pull request closed without merging**